src/common/tree_renderer/graphviz_tree_renderer.cpp

```diff
@@ -0,0 +1,108 @@
+#include "duckdb/common/tree_renderer/graphviz_tree_renderer.hpp"
+
+#include "duckdb/common/pair.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/execution/operator/aggregate/physical_hash_aggregate.hpp"
+#include "duckdb/execution/operator/join/physical_delim_join.hpp"
+#include "duckdb/execution/operator/scan/physical_positional_scan.hpp"
+#include "duckdb/execution/physical_operator.hpp"
+#include "duckdb/parallel/pipeline.hpp"
+#include "duckdb/planner/logical_operator.hpp"
+#include "duckdb/main/query_profiler.hpp"
+#include "utf8proc_wrapper.hpp"
+
+#include <sstream>
+
+namespace duckdb {
+
+string GRAPHVIZTreeRenderer::ToString(const LogicalOperator &op) {
+	std::stringstream ss;
+	Render(op, ss);
+	return ss.str();
+}
+
+string GRAPHVIZTreeRenderer::ToString(const PhysicalOperator &op) {
+	std::stringstream ss;
+	Render(op, ss);
+	return ss.str();
+}
+
+string GRAPHVIZTreeRenderer::ToString(const ProfilingNode &op) {
+	std::stringstream ss;
+	Render(op, ss);
+	return ss.str();
+}
+
+string GRAPHVIZTreeRenderer::ToString(const Pipeline &op) {
+	std::stringstream ss;
+	Render(op, ss);
+	return ss.str();
+}
+
+void GRAPHVIZTreeRenderer::Render(const LogicalOperator &op, std::ostream &ss) {
+	auto tree = RenderTree::CreateRenderTree(op);
+	ToStream(*tree, ss);
+}
+
+void GRAPHVIZTreeRenderer::Render(const PhysicalOperator &op, std::ostream &ss) {
+	auto tree = RenderTree::CreateRenderTree(op);
+	ToStream(*tree, ss);
+}
+
+void GRAPHVIZTreeRenderer::Render(const ProfilingNode &op, std::ostream &ss) {
+	auto tree = RenderTree::CreateRenderTree(op);
+	ToStream(*tree, ss);
+}
+
+void GRAPHVIZTreeRenderer::Render(const Pipeline &op, std::ostream &ss) {
+	auto tree = RenderTree::CreateRenderTree(op);
+	ToStream(*tree, ss);
+}
+
+void GRAPHVIZTreeRenderer::ToStream(RenderTree &root, std::ostream &ss) {
+	const string digraph_format = R"(
+digraph G {
+    node [shape=box, style=rounded, fontname="Courier New", fontsize=10];
```

Comment:
Amazing!

Only question: should this be customizable? Say via a paramether, so that we provide a default but people can override it?

I guess when embedding this you might want to uniform to whatever style you use.
Then likely this can also be done in postprocessing, or in subsequent PR.

Reply:
Yea fair point, not really sure where this should be exposed though

Reply:
Then follow up, if/when this comes up?

Reply:
I guess this also ties in to <https://github.com/duckdb/duckdb/pull/12716#issuecomment-2202648657> to provide a way to set custom configuration options for each format